### PR TITLE
[13.0][FIX] account_financial_report: default company id is always the lowest ID

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
move company_id to the abstract model and correct default definition.

@ForgeFlow

### CODE REVIEW

**Review by JoanSForgeFlow — APPROVED** [review]
LGTM :+1:

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
@JoanSForgeFlow can you backport to 12.0?

**Review by HaraldPanten — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
@JordiBForgeFlow @JoanSForgeFlow I don't think it is needed in v12, `self.env.user.company_id` is fine in v12.

**Comment by LoisRForgeFlow** [comment]
I aborted the merge. I need to include some more changes.

**Comment by LoisRForgeFlow** [comment]
it should be good now.

File: account_financial_report/wizard/general_ledger_wizard.py [hunk 1]
```diff
@@ -119,8 +113,8 @@ def on_change_account_range(self):
     def _init_date_from(self):
         """set start date to begin of current year if fiscal year running"""
         today = fields.Date.context_today(self)
-        last_fsc_month = self.env.user.company_id.fiscalyear_last_month
-        last_fsc_day = self.env.user.company_id.fiscalyear_last_day
+        last_fsc_month = self.env.company.fiscalyear_last_month
```
Comment (pedrobaeza) [inline account_financial_report/wizard/general_ledger_wizard.py]: Shouldn't this be self.company_id?

Comment (LoisRForgeFlow) [reply account_financial_report/wizard/general_ledger_wizard.py]: The problem is that company_id isn't required, but maybe we could use it and fallback to self.env.company which will be more correct. I'm changing it now.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**